alphastats/llm/llm_integration.py

```diff
@@ -186,25 +192,29 @@ def execute_function(
                 function := {
                     "get_gene_function": get_gene_function,
                     "get_enrichment_data": get_enrichment_data,
-                    "perform_dimensionality_reduction": perform_dimensionality_reduction,
                 }.get(function_name)
             ) is not None:
                 return function(**function_args)
 
-            # special treatment for this one
+            # special treatment for these functions
+            elif function_name == "perform_dimensionality_reduction":
+                # TODO add API in dataset
+                perform_dimensionality_reduction(self._dataset, **function_args)
+
             elif function_name == "plot_intensity":
+                # TODO move this logic to dataset
```

Comment:
This relates to my comment on the previous pull request. This logic is not needed, if protein ids are passed, as it was (by name) the case before refactor llm iii.

Reply:
is moved to dataset later